src/TypeSpec/__Private/DictSpec.hack

```diff
@@ -61,4 +61,14 @@ final class DictSpec<Tk as arraykey, Tv> extends TypeSpec<dict<Tk, Tv>> {
       ($k, $_v) ==> $this->tsk->withTrace($kt)->assertType($k),
     );
   }
+
+  <<__Override>>
+  public function toString(): string {
+    return Str\format(
+      '%s<%s, %s>',
+      dict::class,
```

Comment:
Does this expand to `HH\dict` and do we want that? Given that we generally recommend people to not use the namespace for autoloaded classes (the namespace might also be inconsistent across HHVM versions).

Same comment may apply to other specs below (collections...)

Reply:
Yes, I'm doing this intentionally in https://github.com/hhvm/type-assert/pull/34/commits/bdba9adeaafe6f4378003b7ee6b693fc3684f504

The idea is to make it so that as little special-casing is needed to use this as possible - "just add `\` at the start" in codegen.

That said, it's not perfect (e.g. shapes, tuples, resources need special-casing), and I'm not certain it's the right way to go